packages/utils/upgrade/src/modules/version/range.ts

```diff
@@ -8,16 +8,24 @@ export const rangeFactory = (range: string): Version.Range => {
 };
 
 export const rangeFromReleaseType = (current: Version.SemVer, identifier: Version.ReleaseType) => {
-  const fromCurrentTo = (version: Version.LiteralVersion) => {
-    return rangeFactory(`>${current.raw} <=${version}`);
-  };
-
   switch (identifier) {
     case Version.ReleaseType.Major: {
       // semver.inc(_, 'major') will target <major + 1>.0.0 which is what we want
       // e.g. for 4.15.4, it'll return 5.0.0
       const nextMajor = semver.inc(current, 'major') as Version.LiteralSemVer;
-      return fromCurrentTo(nextMajor);
+      return rangeFactory(`>${current.raw} <=${nextMajor}`);
+    }
+    case Version.ReleaseType.Patch: {
+      // This will return the premionr for the next minor
+      // e.g. for 4.15.4, it'll return 4.16.0-0
```

Comment:
That's annoying, the `Version.LiteralSemVer` tells us that the value is of type `<number>.<number>.<number>`, but we have a pre-release identifier in our variable.

I guess you could either remove the cast or modify the type to fit the value.

Reply:
On the same note, why are we using pre-releases instead of regular releases here? Is it to handle special cases?
Why couldn't we do inc('minor') instead?

Reply:
just guessing from my part but the the pre-releases would be part of the upgrade range, if we ever do a pre-release

Reply:
ok never mind it only includes pre-releases if we add it as an option